### ISSUE TITLE
Audit for World of Warcraft 8.x

### ISSUE THREAD

**Issue by spiralofhope** [spiralofhope/_DevPad.GUI#24]

**Comment by ssateneth** [comment]
I'm happy to see you keeping an eye on this addon! I was just about to create my own issue thread about 8.x compatability when I saw this one already existing.

Anywho, BFA and the new API is now on PTR so you can code for it without needing beta access. Just a few errors so far when loading in.

<details>
  <summary>Click to expand</summary>

```
Message: ...\AddOns\_DevPad\Libs\AceComm-3.0\ChatThrottleLib.lua:216: hooksecurefunc(): SendAddonMessage is not a function
Time: Thu Jun 14 19:33:10 2018
Count: 1
Stack: ...\AddOns\_DevPad\Libs\AceComm-3.0\ChatThrottleLib.lua:216: hooksecurefunc(): SendAddonMessage is not a function
[C]: in function `hooksecurefunc'
...\AddOns\_DevPad\Libs\AceComm-3.0\ChatThrottleLib.lua:216: in function `Init'
...\AddOns\_DevPad\Libs\AceComm-3.0\ChatThrottleLib.lua:498: in main chunk

Locals:
```

```
Message: Error loading Interface\AddOns\_DevPad\_DevPad.options.lua
Time: Thu Jun 14 19:33:10 2018
Count: 3
```

```
Message: Interface\AddOns\_DevPad\_DevPad.lua:492: attempt to call global '_DevPad_setup_options_pane' (a nil value)
Time: Thu Jun 14 19:33:10 2018
Count: 1
Stack: Interface\AddOns\_DevPad\_DevPad.lua:492: attempt to call global '_DevPad_setup_options_pane' (a nil value)
Interface\AddOns\_DevPad\_DevPad.lua:492: in function <Interface\AddOns\_DevPad\_DevPad.lua:490>
(tail call): ?

Locals: self = <unnamed> {
 0 = <userdata>
 PLAYER_LOGOUT = <function> defined @Interface\AddOns\_DevPad\_DevPad.lua:514
 ADDON_LOADED = <function> defined @Interface\AddOns\_DevPad\_DevPad.lua:490
 OnEvent = <function> defined @Interface\AddOns\_DevPad\_DevPad.lua:520
}
Event = "ADDON_LOADED"
AddOn = "_DevPad"
(*temporary) = nil
(*temporary) = "attempt to call global '_DevPad_setup_options_pane' (a nil value)"
AddOnName = "_DevPad"
NS = <table> {
 Callbacks = <table> {
 }
 ReceiveQueue = <table> {
 }
 Print = <function> defined @Interface\AddOns\_DevPad\_DevPad.lua:81
 SlashCommand = <function> defined @Interface\AddOns\_DevPad\_DevPad.lua:528
 FolderRoot = <table> {
 }
 L = <table> {
 }
 UnregisterAllCallbacks = <function> defined @Interface\AddOns\_DevPad\Libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:202
 GetAbsObject = <function> defined @Interface\AddOns\_DevPad\_DevPad.lua:436
 IterateScripts = <function> defined @Interface\AddOns\_DevPad\_DevPad.lua:411
 SafeCall = <function> defined @Interface\AddOns\_DevPad\_DevPad.lua:97
 RegisterCallback = <function> defined @Interface\AddOns\_DevPad\Libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:118
 OnCommReceived = <function> defined @Interface\AddOns\_DevPad\_DevPad.lua:449
 UnregisterCallback = <function> defined @Interface\AddOns\_DevPad\Libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:181
 DefaultScripts = <table> {
 }
 GetClass = <function> defined @Interface\AddOns\_DevPad\_DevPad.lua:190
 Frame = <unnamed> {
 }
 ReceiveIgnored = <table> {
 }
 FindScripts = <function> defined @Interface\AddOns\_DevPad\_DevPad.lua:424
}
AceComm = <table> {
 OnEmbedDisable = <function> defined @Interface\AddOns\_DevPad\Libs\AceComm-3.0\AceComm-3.0.lua:375
 embeds = <table> {
 }
 multipart_reassemblers = <table> {
 }
 frame = AceComm30Frame {
 }
 callbacks = <table> {
 }
 OnReceiveMultipartNext = <function> defined @Interface\AddOns\_DevPad\Libs\AceComm-3.0\AceComm-3.0.lua:211
 RegisterComm = <function> defined @Interface\AddOns\_DevPad\Libs\AceComm-3.0\AceComm-3.0.lua:63
 UnregisterComm = <function> defined @Interface\AddOns\_DevPad\Libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:181
 multipart_spool = <table> {
 }
 _RegisterComm = <function> defined @Interface\AddOns\_DevPad\Libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:118
 OnReceiveMultipartLast = <function> defined @Interface\AddOns\_DevPad\Libs\AceComm-3.0\AceComm-3.0.lua:232
 SendCommMessage = <function> defined @Interface\AddOns\_DevPad\Libs\AceComm-3.0\AceComm-3.0.lua:87
 multipart_origprefixes = <table> {
 }
 OnReceiveMultipartFirst = <function> defined @Interface\AddOns\_DevPad\Libs\AceComm-3.0\AceComm-3.0.lua:197
 Embed = <function> defined @Interface\AddOns\_DevPad\Libs\AceComm-3.0\AceComm-3.0.lua:367
 UnregisterAllComm = <function> defined @Interface\AddOns\_DevPad\Libs\CallbackHandler-1.0\CallbackHandler-1.0.lua:202
}
COMM_PREFIX = "_DP"
```
</details>


And doing `/pad` shows the _devpad GUI but nothing contained in it, as well as a few other lua errors.

<details>
  <summary>Click to expand</summary>

```
Message: Error loading Interface\AddOns\_DevPad.GUI\_DevPad.GUI.List.Search_optional.lua
Time: Thu Jun 14 19:34:29 2018
Count: 3
```

```
Message: Error loading Interface\AddOns\_DevPad.GUI\_DevPad.GUI.Editor.Color_optional.lua
Time: Thu Jun 14 19:34:29 2018
Count: 3
```

```
Message: Error loading Interface\AddOns\_DevPad.GUI\_DevPad.GUI.Editor.History_optional.lua
Time: Thu Jun 14 19:34:29 2018
Count: 3
```

```
Message: Error loading Interface\AddOns\_DevPad.GUI\_DevPad.GUI.Editor.LineNumbers_optional.lua
Time: Thu Jun 14 19:34:29 2018
Count: 3
```

https://i.imgur.com/NfccTTv.png

</details>

Also, will the issue with scaling be revisited I hope? Maybe blizzard fixed the textbox scaling misaligning thing

**Comment by spiralofhope** [comment]
I'm not seeing any errors on the beta.  I'll do more aggressive testing.

**Comment by ssateneth** [comment]
It may have had something to do with an old version or mixed versions. I wiped the folders I had and redownloaded 7.3.2 devpad and GUI. Now I only get these 2 errors.

```
Message: ...\AddOns\_DevPad\Libs\AceComm-3.0\ChatThrottleLib.lua:216: hooksecurefunc(): SendAddonMessage is not a function
Time: Mon Jun 25 01:36:43 2018
Count: 1
Stack: ...\AddOns\_DevPad\Libs\AceComm-3.0\ChatThrottleLib.lua:216: hooksecurefunc(): SendAddonMessage is not a function
[C]: in function `hooksecurefunc'
...\AddOns\_DevPad\Libs\AceComm-3.0\ChatThrottleLib.lua:216: in function `Init'
...\AddOns\_DevPad\Libs\AceComm-3.0\ChatThrottleLib.lua:498: in main chunk

Locals: 
```

```
Message: (null)
Time: Mon Jun 25 01:36:44 2018
Count: 4
```

**Comment by ssateneth** [comment]
On first glance, it seems that _Devpad and its GUI are working fine. The error above probably had to do with outdated Ace libraries. I have standalone Ace library installed as alpha version and haven't had any problems with devpad. This is with using 7.3.0.6 devpad and GUI, which has the old implementation of GUI scaling size too (which also works fine)

**Comment by spiralofhope** [comment]
I also use the standalone bundle of Ace addons.  I'm not sure how all the library stuff even works so I'll probably just note the requirement for now.

**Comment by spiralofhope** [comment]
Whatever Ace issue existed has vanished.

I made #25 - Revisit scaling

**Comment by spiralofhope** [comment]
Fixed #1 Cursor doesn't line up with active line.
in 8.0.1.2
